v2.0-rc.3 When trying to create a Bucket : runtime error: invalid memory address or nil pointer dereference

__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Start a Debian 10 fresh new node
2. Install latest InfluxDB OSS release: v2.0-rc.3, according to the get started page on the official doc
3. Start InfluxDB using `influxd --http-bind-address=:9913 --reporting-disabled` _(note that bug happens even with a simple `influxd` command)_
4. Create a new bucket using `POST /api/v2/buckets` (or actually using the Influx UI, same result) with following parameters: 
```
{ "name": "bucket",
"orgID": "${orgId}", 
"retentionRules":[{"everySeconds":302400,"type":"expire"}]}
```
Bucket is created successfully.

5. However, repeating the exact same request, trying to create a new bucket using the same bucket name results in a `500 Internal server error` on the client side. On the server side we get a `panic: runtime error: invalid memory address or nil pointer dereference`. Full logs available bottom of the post.

__Expected behavior:__

Trying to create a bucket with existing name shouldn't return a fatal error but just a simple "bucket already exists" error (behavior I have on previous versions).

__Actual behavior:__

As mentioned previously, just trying to create a bucket with an already existing name results in a 500 error. 
InfluxDB logs available at the bottom of the post. 

__Environment info:__

* System info: 
```
uname -srm
Linux 4.19.0-12-amd64 x86_64
```

* InfluxDB version: 
`InfluxDB 2.0.0-rc.3 (git: f46a3bd91e) build_date: 2020-10-29T22:17:55Z`

__Config:__
I didn't touch configuration.

<!-- The following sections are only required if relevant. -->

__Logs:__

 ```
panic: runtime error: invalid memory address or nil pointer dereference
 -> go.uber.org/zap.(*Logger).check
 ->   /go/pkg/mod/go.uber.org/zap@v1.14.1/logger.go:263

    go.uber.org/zap.(*Logger).Error
      /go/pkg/mod/go.uber.org/zap@v1.14.1/logger.go:202
    github.com/influxdata/influxdb/v2/storage.(*BucketService).CreateBucket.func1
      /go/src/github.com/influxdata/influxdb/storage/bucket_service.go:49
    github.com/influxdata/influxdb/v2/storage.(*BucketService).CreateBucket
      /go/src/github.com/influxdata/influxdb/storage/bucket_service.go:55
    github.com/influxdata/influxdb/v2/tenant.(*AuthedBucketService).CreateBucket
      /go/src/github.com/influxdata/influxdb/tenant/middleware_bucket_auth.go:95
    github.com/influxdata/influxdb/v2/tenant.(*BucketHandler).handlePostBucket
      /go/src/github.com/influxdata/influxdb/tenant/http_server_bucket.go:273
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi.(*Mux).routeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:431
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi.(*Mux).ServeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:70
    github.com/go-chi/chi.(*Mux).Mount.func1
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:298
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi.(*Mux).routeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:431
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi/middleware.RealIP.func1
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/middleware/realip.go:34
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi/middleware.RequestID.func1
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/middleware/request_id.go:76
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi/middleware.Recoverer.func1
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/middleware/recoverer.go:37
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi.(*Mux).ServeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:70
    github.com/go-chi/chi.(*Mux).Mount.func1
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:298
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi.(*Mux).routeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:431
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/influxdata/influxdb/v2/kit/transport/http.SetCORS.func1
      /go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:34
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi/middleware.StripSlashes.func1
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/middleware/strip.go:25
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/influxdata/influxdb/v2/kit/transport/http.SkipOptions.func1
      /go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:72
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/influxdata/influxdb/v2/http.panicMW.func1.1
      /go/src/github.com/influxdata/influxdb/http/router.go:126
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi.(*Mux).ServeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:70
    github.com/influxdata/influxdb/v2/kit/feature.(*Handler).ServeHTTP
      /go/src/github.com/influxdata/influxdb/kit/feature/middleware.go:43
    github.com/influxdata/influxdb/v2/http.(*AuthenticationHandler).ServeHTTP
      /go/src/github.com/influxdata/influxdb/http/authentication_middleware.go:125
    github.com/influxdata/influxdb/v2/kit/transport/http.SetCORS.func1
      /go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:34
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/influxdata/influxdb/v2/kit/transport/http.SkipOptions.func1
      /go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:72
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/influxdata/influxdb/v2/http.(*PlatformHandler).ServeHTTP
      /go/src/github.com/influxdata/influxdb/http/platform_handler.go:78
    github.com/go-chi/chi.(*Mux).Mount.func1
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:298
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/influxdata/influxdb/v2/kit/transport/http.Metrics.func1.1
      /go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:57
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/influxdata/influxdb/v2/kit/transport/http.Trace.func1.1
      /go/src/github.com/influxdata/influxdb/kit/transport/http/middleware.go:97
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi.(*ChainHandler).ServeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/chain.go:31
    github.com/go-chi/chi.(*Mux).routeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:431
    net/http.HandlerFunc.ServeHTTP
      /usr/local/go/src/net/http/server.go:2042
    github.com/go-chi/chi.(*Mux).ServeHTTP
      /go/pkg/mod/github.com/go-chi/chi@v4.1.0+incompatible/mux.go:86
    github.com/influxdata/influxdb/v2/http.(*Handler).ServeHTTP
      /go/src/github.com/influxdata/influxdb/http/handler.go:143
    net/http.serverHandler.ServeHTTP
      /usr/local/go/src/net/http/server.go:2843
    net/http.(*conn).serve
      /usr/local/go/src/net/http/server.go:1925
    created by net/http.(*Server).Serve
      /usr/local/go/src/net/http/server.go:2969
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -4,6 +4,7 @@
 
 1. [19856](https://github.com/influxdata/influxdb/pull/19856): make tagKeys and tagValues work for edge cases involving _field
 1. [19853](https://github.com/influxdata/influxdb/pull/19853): Use updated HTTP client for authorization service
+1. [19886](https://github.com/influxdata/influxdb/pull/19886): Add Logger to constructor function to ensure log field is initialized
 
 ## v2.0.0-rc.3 [2020-10-29]
 

--- a/cmd/influxd/launcher/launcher.go
+++ b/cmd/influxd/launcher/launcher.go
@@ -1122,7 +1122,7 @@ func (m *Launcher) run(ctx context.Context) (err error) {
 		labelSvc = label.NewLabelController(m.flagger, m.kvService, ls)
 	}
 
-	ts.BucketService = storage.NewBucketService(ts.BucketService, m.engine)
+	ts.BucketService = storage.NewBucketService(m.log, ts.BucketService, m.engine)
 	ts.BucketService = dbrp.NewBucketService(m.log, ts.BucketService, dbrpSvc)
 
 	var onboardOpts []tenant.OnboardServiceOptionFn

--- a/cmd/influxd/upgrade/upgrade.go
+++ b/cmd/influxd/upgrade/upgrade.go
@@ -470,7 +470,7 @@ func newInfluxDBv2(ctx context.Context, opts *optionsV2, log *zap.Logger) (svc *
 		storage.WithMetaClient(svc.meta),
 	)
 
-	svc.ts.BucketService = storage.NewBucketService(svc.ts.BucketService, engine)
+	svc.ts.BucketService = storage.NewBucketService(log, svc.ts.BucketService, engine)
 	// on-boarding service (influx setup)
 	svc.onboardSvc = tenant.NewOnboardService(svc.ts, authSvc)
 

--- a/storage/bucket_service.go
+++ b/storage/bucket_service.go
@@ -28,9 +28,10 @@ type BucketService struct {
 
 // NewBucketService returns a new BucketService for the provided EngineSchema,
 // which typically will be an Engine.
-func NewBucketService(s influxdb.BucketService, engine EngineSchema) *BucketService {
+func NewBucketService(log *zap.Logger, s influxdb.BucketService, engine EngineSchema) *BucketService {
 	return &BucketService{
 		BucketService: s,
+		log:           log,
 		engine:        engine,
 	}
 }

--- a/storage/bucket_service_test.go
+++ b/storage/bucket_service_test.go
@@ -26,7 +26,9 @@ func TestBucketService(t *testing.T) {
 	engine := mocks.NewMockEngineSchema(ctrl)
 
 	inmemService := newInMemKVSVC(t)
-	service := storage.NewBucketService(inmemService, engine)
+
+	logger := zaptest.NewLogger(t)
+	service := storage.NewBucketService(logger, inmemService, engine)
 
 	if err := service.DeleteBucket(context.TODO(), *i); err == nil {
 		t.Fatal("expected error, got nil")
@@ -45,7 +47,7 @@ func TestBucketService(t *testing.T) {
 	engine.EXPECT().DeleteBucket(gomock.Any(), org.ID, bucket.ID)
 
 	// Test deleting a bucket calls into the deleter.
-	service = storage.NewBucketService(inmemService, engine)
+	service = storage.NewBucketService(logger, inmemService, engine)
 
 	if err := service.DeleteBucket(context.TODO(), bucket.ID); err != nil {
 		t.Fatal(err)

--- a/task/backend/analytical_storage_test.go
+++ b/task/backend/analytical_storage_test.go
@@ -64,7 +64,7 @@ func TestAnalyticalStore(t *testing.T) {
 				svcStack = backend.NewAnalyticalRunStorage(logger, svc, ts.BucketService, svc, rr, ab.QueryService())
 			)
 
-			ts.BucketService = storage.NewBucketService(nil, ts.BucketService, ab.storageEngine)
+			ts.BucketService = storage.NewBucketService(logger, ts.BucketService, ab.storageEngine)
 
 			authCtx := icontext.SetAuthorizer(ctx, &influxdb.Authorization{
 				Permissions: influxdb.OperPermissions(),

--- a/task/backend/analytical_storage_test.go
+++ b/task/backend/analytical_storage_test.go
@@ -64,7 +64,7 @@ func TestAnalyticalStore(t *testing.T) {
 				svcStack = backend.NewAnalyticalRunStorage(logger, svc, ts.BucketService, svc, rr, ab.QueryService())
 			)
 
-			ts.BucketService = storage.NewBucketService(ts.BucketService, ab.storageEngine)
+			ts.BucketService = storage.NewBucketService(nil, ts.BucketService, ab.storageEngine)
 
 			authCtx := icontext.SetAuthorizer(ctx, &influxdb.Authorization{
 				Permissions: influxdb.OperPermissions(),

